sklearn/utils/estimator_checks.py

```diff
@@ -687,7 +691,11 @@ def check_fit_score_takes_y(name, Estimator):
 @ignore_warnings
 def check_estimators_dtypes(name, Estimator):
     rnd = np.random.RandomState(0)
-    X_train_32 = 3 * rnd.uniform(size=(20, 5)).astype(np.float32)
+    if name in ["BoxCoxTransformer", "SpectralCoclustering",
```

Comment:
The array being randomly generated contains 0s which is causing ``inf`` and ``nan`` which results in ``ValueError`` due to [this](https://github.com/maniteja123/scikit-learn/blob/bf459f445c43a663df565bc0aa237f57acfc7264/sklearn/cluster/bicluster.py#L170). The only approach was to generate positive numbers. Is there something I am doing wrong here ?